clang/7+cuda/9 build -Werror-unused parameter error in nightly test

Sample error:

```
18:48:10 /home/jenkins/slave/workspace/Kokkos_develop_kokkos-dev_clang_701_cuda_92/kokkos/core/src/impl/Kokkos_Atomic_Load.hpp:123:8: error: unused parameter 'ptr' [-Werror,-Wunused-parameter]
18:48:10     T* ptr, typename std::enable_if<!(sizeof(T) == 1 || sizeof(T) == 2 ||
18:48:10        ^
18:48:10 1 error generated when compiling for sm_35.
```

Sample reproducer:
```
   #   Load modules:
         module load sems-env sems-cmake/3.12.2 kokkos-env kokkos-hwloc/1.10.1/base sems-clang/7.0.1 sems-cuda/9.2
 
   #   Use generate_makefile line below to call cmake which generates makefile for this build:
         /home/jenkins/slave/workspace/Kokkos_develop_kokkos-dev_clang_701_cuda_92/kokkos/generate_makefile.bash --with-devices=Cuda,OpenMP --arch=Kepler35 --compiler=/projects/sems/install/rhel6-x86_64/sems/compiler/clang/7.0.1/base/bin/clang++ --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized " --cxxstandard="11" --ldflags="" --with-cuda=/projects/sems/install/rhel6-x86_64/sems/compiler/cuda/9.2/base --kokkos-path=/home/jenkins/slave/workspace/Kokkos_develop_kokkos-dev_clang_701_cuda_92/kokkos --no-examples 
```

--- a/core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp
+++ b/core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp
@@ -260,6 +260,8 @@ struct CudaParallelLaunch<
             (prefer_shmem ? cudaFuncCachePreferShared
                           : cudaFuncCachePreferL1)));
       }
+#else
+      (void)prefer_shmem;
 #endif
 
       // Copy functor to constant memory on the device
@@ -317,6 +319,8 @@ struct CudaParallelLaunch<DriverType, Kokkos::LaunchBounds<0, 0>,
             (prefer_shmem ? cudaFuncCachePreferShared
                           : cudaFuncCachePreferL1)));
       }
+#else
+      (void)prefer_shmem;
 #endif
 
       // Copy functor to constant memory on the device
@@ -371,6 +375,8 @@ struct CudaParallelLaunch<
             (prefer_shmem ? cudaFuncCachePreferShared
                           : cudaFuncCachePreferL1)));
       }
+#else
+      (void)prefer_shmem;
 #endif
 
       KOKKOS_ENSURE_CUDA_LOCK_ARRAYS_ON_DEVICE();
@@ -419,6 +425,8 @@ struct CudaParallelLaunch<DriverType, Kokkos::LaunchBounds<0, 0>,
             (prefer_shmem ? cudaFuncCachePreferShared
                           : cudaFuncCachePreferL1)));
       }
+#else
+      (void)prefer_shmem;
 #endif
 
       KOKKOS_ENSURE_CUDA_LOCK_ARRAYS_ON_DEVICE();
@@ -465,6 +473,8 @@ struct CudaParallelLaunch<
             (prefer_shmem ? cudaFuncCachePreferShared
                           : cudaFuncCachePreferL1)));
       }
+#else
+      (void)prefer_shmem;
 #endif
 
       KOKKOS_ENSURE_CUDA_LOCK_ARRAYS_ON_DEVICE();
@@ -516,6 +526,8 @@ struct CudaParallelLaunch<DriverType, Kokkos::LaunchBounds<0, 0>,
             (prefer_shmem ? cudaFuncCachePreferShared
                           : cudaFuncCachePreferL1)));
       }
+#else
+      (void)prefer_shmem;
 #endif
 
       KOKKOS_ENSURE_CUDA_LOCK_ARRAYS_ON_DEVICE();

--- a/core/src/impl/Kokkos_Atomic_Load.hpp
+++ b/core/src/impl/Kokkos_Atomic_Load.hpp
@@ -125,8 +125,7 @@ __device__ __inline__ T _relaxed_atomic_load_impl(
                                     void const**>::type = nullptr) {
   T rv{};
   // TODO remove a copy operation here?
-  Kokkos::Impl::atomic_oper_fetch(NoOpOper<T>{}, &rv, rv);
-  return rv;
+  return Kokkos::Impl::atomic_oper_fetch(NoOpOper<T>{}, ptr, rv);
 }
 
 template <class T>

